pkg/provider/azure_loadbalancer.go

```diff
@@ -2621,6 +2621,7 @@ func (az *Cloud) reconcileSecurityRules(sg network.SecurityGroup, service *v1.Se
 				}
 				if sharedRule.DestinationAddressPrefixes == nil {
 					klog.V(4).Infof("Didn't find DestinationAddressPrefixes in shared rule for service %s", service.Name)
+					updatedRules = append(updatedRules[:sharedIndex], updatedRules[sharedIndex+1:]...)
```

Comment:
Why do we need to include it here and remove it later in L2671?

Reply:
we removed the rules here..